lib/transitive.ts

```diff
@@ -0,0 +1,841 @@
+  /**
+   * Custom styling rules that affect rendering behavior
+   */
+  styles: TransitiveStyles
+  /**
+   * Whether to enable the display's built-in zoom/pan functionality (defaults
+   * to true)
+   */
+  zoomEnabled?: boolean
+  /**
+   * A list of different styling configurations to show at various zoom levels.
+   * This list of Zoomfactors must be ordered by each object's "minScale" key
+   * with the lowest "minScale" value appearing first and the largest one
+   * appearing last. There must be a "minScale" value below 1 in the list of
+   * definitions.
+   *
+   * Default values for this config item are used, unless overridden by adding
+   * this key and value. The default values can be found here:
+   * https://github.com/conveyal/transitive.js/blob/6a8932930de003788b9034609697421731e7f812/lib/display/display.js#L78-L92
+   */
+  zoomFactors?: ZoomFactor[]
+}
+
+/**
+ * No clue what this global Display thing is. :(
+ */
+declare class Display {
+  constructor(arg0: Transitive)
+}
+
+/**
+ * A transformation {x, y, k} to the *initial* state of the map., where
+ * (x, y) is the pixel offset and k is a scale factor relative to an initial
+ * zoom level of 1.0. Intended primarily to support D3-style panning/zooming.
+ */
+type DisplayTransform = {
+  k: number
+  x: number
+  y: number
+}
+
+// FIXME add better typing once more files are typed
+type Journeys = { [id: string]: { path: Record<string, unknown> } }
+
+export default class Transitive {
+  data: TransitiveData | null | undefined
+  display!: CanvasDisplay | SvgDisplay
+  el?: HTMLElement
+  // FIXME: somehow typing the emit method (which is injected at the end of this
+  // file by component-emitter causes the emit method to not work.
+  // emit!: (message: string, transitiveInstance: this, el?: HTMLElement) => void
+  labeler!: Labeler
+  options?: TransitiveOptions
+  network: Network | null | undefined
+  renderer!: DefaultRenderer | WireframeRenderer
+  styler!: Styler
+
+  constructor(options: TransitiveOptions) {
+    if (!(this instanceof Transitive)) return new Transitive(options)
+
+    this.options = options
+    if (this.options.zoomEnabled === undefined) this.options.zoomEnabled = true
+    if (this.options.autoResize === undefined) this.options.autoResize = true
+    if (this.options.groupEdges === undefined) this.options.groupEdges = true
+
+    if (options.el) this.el = options.el
+
+    this.display =
+      this.options.display === 'canvas'
+        ? new CanvasDisplay(this)
+        : new SvgDisplay(this)
+
+    this.data = options.data
+
+    this.setRenderer(this.options.initialRenderer || 'default')
+
+    this.labeler = new Labeler(this)
+    this.styler = new Styler(options.styles, this)
+
+    if (options.initialBounds) {
+      this.display.fitToWorldBounds([
+        sm.forward(options.initialBounds[0]),
+        sm.forward(options.initialBounds[1])
+      ])
+    }
+  }
+
+  /**
+   * Clear the Network data and redraw the (empty) map
+   */
+
```

Comment:
Remove blank line between comments and method declaration for all methods below.